internal/oci/runtime_oci.go

```diff
@@ -905,6 +905,11 @@ func (r *runtimeOCI) DeleteContainer(ctx context.Context, c *Container) error {
 		return nil
 	}
 
+	if c.state.OOMKilled {
```

Comment:
You could remove the if branch since DeleteLabelValues() should be a no-op when the label value doesn't exist.

Reply:
will it skim through all the values to check though? I figured we could avoid the lookup because we already know whether it's present or not

Reply:
I don't know in depth the internals of prometheus/client_golang but IIUC a MetricVec indexes series by their label hash so the lookup should be fast (https://github.com/prometheus/client_golang/blob/046e3203b61d1b5000b162e210a91975f098d742/prometheus/vec.go#L319).

Reply:
thanks for the suggestion, I think I prefer checking the bool as it'll be the fastest and doesn't increase the complexity too much